File: Graphs Demo/Graphs Demo/Edge.cs
namespace Graphs_Demo
{
    using System;

    public class Edge
    {
        private Node target;

        public Edge(Node target, int weight)
        {
            this.Target = target;
            this.Weight = weight;
        }

        public Node Target
        {
            get
            {
                return this.target;
            }

            private set
            {
                if (value == null)
                {
                    throw new ArgumentNullException("Target cannot be set to null!");
                }

                this.target = value;
            }
        }

        public int Weight { get; private set; }
    }
}

File: Graphs Demo C++/src/Edge.cpp
#include "Edge.h"

Edge::Edge(Node& target, int weight) : target(target)
{
    this->weight = weight;
}

Edge::~Edge()
{
}

Edge::Edge(const Edge& other) : target(other.target)
{
    this->weight = other.weight;
}

Edge& Edge::operator=(const Edge& other)
{
    if (this == &other)
    {
        //this->target = other.target;
    }

    return *this;
}

Node& Edge::getTarget() const
{
    return this->target;
}

int Edge::getWeight() const
{
    return this->weight;
}
